python/tvm/target/target.py

```diff
@@ -443,22 +443,45 @@ def intel_graphics(model="unknown", options=None):
 
 
 MICRO_SUPPORTED_MODELS = {
+    # X86 emulation
     "host": [],
-    "atsamd51": ["-mcpu=cortex-m4"],
-    "cxd5602gg": ["-mcpu=cortex-m4"],
+    # STM High-Performance
+    "stm32h7xx": ["-keys=arm_cpu,cpu", "-mcpu=cortex-m7", "-march=armv7e-m"],
+    "stm32f7xx": ["-keys=arm_cpu,cpu", "-mcpu=cortex-m7", "-march=armv7e-m"],
+    "stm32f4xx": ["-keys=arm_cpu,cpu", "-mcpu=cortex-m4"],
+    "stm32f2xx": ["-keys=arm_cpu,cpu", "-mcpu=cortex-m3"],
+    # STM Mainstream
+    "stm32g0xx": ["-keys=arm_cpu,cpu", "-mcpu=cortex-m0+"],
+    "stm32f0xx": ["-keys=arm_cpu,cpu", "-mcpu=cortex-m0"],
+    "stm32f1xx": ["-keys=arm_cpu,cpu", "-mcpu=cortex-m3"],
+    "stm32g4xx": ["-keys=arm_cpu,cpu", "-mcpu=cortex-m4"],
+    "stm32f3xx": ["-keys=arm_cpu,cpu", "-mcpu=cortex-m4"],
+    "stm32l4r5xx": ["-keys=arm_cpu,cpu", "-mcpu=cortex-m4"],
+    # STM Low-power
+    "stm32u5xx": ["-keys=arm_cpu,cpu", "-mcpu=cortex-m33"],
+    "stm32l5xx": ["-keys=arm_cpu,cpu", "-mcpu=cortex-m33"],
+    "stm32l4xx": ["-keys=arm_cpu,cpu", "-mcpu=cortex-m4"],
+    "stm32l1xx": ["-keys=arm_cpu,cpu", "-mcpu=cortex-m3"],
+    "stm32l0xx": ["-keys=arm_cpu,cpu", "-mcpu=cortex-m0+"],
+    # NRF
+    "nrf52840": ["-keys=arm_cpu,cpu", "-mcpu=cortex-m4+nodsp"],
+    "nrf5340dk": ["-keys=arm_cpu,cpu", "-mcpu=cortex-m33"],
+    # Microchip
+    "atsamd51": ["-keys=arm_cpu,cpu", "-mcpu=cortex-m4"],
+    "sam3x8e": ["-keys=arm_cpu,cpu", "-mcpu=cortex-m3"],
+    # Sony
+    "cxd5602gg": ["-keys=arm_cpu,cpu", "-mcpu=cortex-m4"],
+    # ESP
     "esp32": [],
-    "imxrt10xx": ["-mcpu=cortex-m7"],
-    "mps2_an521": ["-mcpu=cortex-m33"],
-    "mps3_an547": ["-mcpu=cortex-m55"],
-    "nrf52840": ["-mcpu=cortex-m4+nodsp"],
-    "nrf5340dk": ["-mcpu=cortex-m33"],
-    "rp2040": ["-mcpu=cortex-m0"],
-    "sam3x8e": ["-mcpu=cortex-m3"],
-    "stm32f746xx": ["-mcpu=cortex-m7", "-march=armv7e-m"],
-    "stm32h7xx": ["-mcpu=cortex-m7"],
-    "stm32l4r5zi": ["-mcpu=cortex-m4"],
-    "stm32u5xx": ["-mcpu=cortex-m33"],
-    "zynq_mp_r5": ["-mcpu=cortex-r5"],
+    # NXP
+    "imxrt10xx": ["-keys=arm_cpu,cpu", "-mcpu=cortex-m7"],
```

Comment:
Does this also need "-march=armv7e-m"?

Reply:
I think `armv7e-m` should be on all cortex-m4 and cortex-m7 boards:
https://en.wikipedia.org/wiki/ARM_Cortex-M

Reply:
cc @areusch

Reply:
i believe the compiler infers this from -mcpu for ARM parts. cc @Mousius if he knows more

Reply:
@areusch that was my expectation too, but I saw `-march=armv7e-m` in one of the targets and I got confused.

Reply:
+1 - that's why I asked, because I saw it specified above.  Maybe that can be removed?